cilium-cli/connectivity/check/logging.go

```diff
@@ -126,11 +128,15 @@ func (ct *ConnectivityTest) LogOwners(scenario ownedScenario) {
 	}
 
 	var workflowOwners []codeowners.Owner
+	var ghWorkflow string
 	// Example: cilium/cilium/.github/workflows/conformance-kind-proxy-embedded.yaml@refs/pull/37593/merge
-	ghWorkflow := os.Getenv("GITHUB_WORKFLOW_REF")
-	ghWorkflow, _, _ = strings.Cut(ghWorkflow, "@")
-	segments := strings.Split(ghWorkflow, "/")
-	ghWorkflow = strings.Join(segments[2:], "/")
+	ghWorkflowRef := os.Getenv("GITHUB_WORKFLOW_REF")
+	matches := ghWorkflowRegexp.FindStringSubmatch(ghWorkflowRef)
+	// here matches should either be nil (no match) or a slice with two values:
+	// the full match and the capture.
+	if len(matches) == 2 {
+		ghWorkflow = matches[1]
```

Comment:
Naive question, how do you tell that the capture omits the initial `org/repo/` match?

Reply:
in `^(?:.+?)/(?:.+?)/(.+?)@.*$`:
* `^` start of line anchor
* `(?:.+?)/(?:.+?)/` is the `org/repo` part and it is using the non-capturing group syntax `(?:...)`
* `(.+?)` is a non-eager group, so we'll stop before the first `@`. this is what we're interested in, i.e. `ghWorkflow`, so it's a capture group.
* `@.*$` matches the rest of the line, at least `cilium/cilium/.github/workflows/conformance-kind-proxy-embedded.yaml@` will still match (i.e. nothing after `@`) but I think it's ok.

So when we have a match, `matches[0]` will always be the full `os.Getenv("GITHUB_WORKFLOW_REF")` because our regexp is anchored both at start and EOL, and `matches[1]` will be our only regexp capture group.

Reply:
Looks like the answer lies in the use of `(?:...) syntax, which matches but does not capture that submatch :+1:

Reply:
Ah thank you for the detailed explanation!